File: General/StressTest/cs/StressTest/OpenCloseTask.cs
using SolidEdgeCommunity;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.InteropServices;
using System.Text;
using System.Threading.Tasks;

namespace StressTest
{
    public class OpenCloseTask : IsolatedTaskProxy
    {
        public void DoOpenClose(string fileName)
        {
            InvokeSTAThread<string>(DoOpenCloseInternal, fileName);
        }

        void DoOpenCloseInternal(string fileName)
        {
            //SolidEdgeFramework.Application application = null;
            SolidEdgeFramework.Documents documents = null;
            SolidEdgeFramework.SolidEdgeDocument document = null;

            // Register with OLE to handle concurrency issues on the current thread.
            SolidEdgeCommunity.OleMessageFilter.Register();

            try
            {
                // Get reference to application object.
                var application = this.Application;

                // Get reference to documents collection.
                documents = application.Documents;

                // Open the document.
                document = (SolidEdgeFramework.SolidEdgeDocument)documents.Open(fileName);

                // Not sure why but as of ST8, I had to add a Sleep() call to prevent crashing...
                System.Threading.Thread.Sleep(500);

                // Do idle processing.
                application.DoIdle();

                // Close the document.
                document.Close(false);

                //System.Threading.Thread.Sleep(500);

                // Do idle processing.
                application.DoIdle();

                documents.Close();
            }
            catch (System.Exception ex)
            {
                Console.WriteLine(ex.Message);
            }
            finally
            {
                SolidEdgeCommunity.OleMessageFilter.Unregister();
            }
        }
    }
}

File: General/StressTest/vb/StressTest/OpenCloseTask.vb
Imports SolidEdgeCommunity
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Runtime.InteropServices
Imports System.Text
Imports System.Threading.Tasks

Namespace StressTest
    Public Class OpenCloseTask
        Inherits IsolatedTaskProxy

        Public Sub DoOpenClose(ByVal fileName As String)
            InvokeSTAThread(Of String)(AddressOf DoOpenCloseInternal, fileName)
        End Sub

        Private Sub DoOpenCloseInternal(ByVal fileName As String)
            'SolidEdgeFramework.Application application = null;
            Dim documents As SolidEdgeFramework.Documents = Nothing
'INSTANT VB NOTE: The variable document was renamed since Visual Basic does not handle local variables named the same as class members well:
            Dim document_Renamed As SolidEdgeFramework.SolidEdgeDocument = Nothing

            ' Register with OLE to handle concurrency issues on the current thread.
            SolidEdgeCommunity.OleMessageFilter.Register()

            Try
                ' Get reference to application object.
'INSTANT VB NOTE: The variable application was renamed since Visual Basic does not handle local variables named the same as class members well:
                Dim application_Renamed = Me.Application

                ' Get reference to documents collection.
                documents = application_Renamed.Documents

                ' Open the document.
                document_Renamed = DirectCast(documents.Open(fileName), SolidEdgeFramework.SolidEdgeDocument)

                ' Not sure why but as of ST8, I had to add a Sleep() call to prevent crashing...
                System.Threading.Thread.Sleep(500)

                ' Do idle processing.
                application_Renamed.DoIdle()

                ' Close the document.
                document_Renamed.Close(False)

                'System.Threading.Thread.Sleep(500);

                ' Do idle processing.
                application_Renamed.DoIdle()

                documents.Close()
            Catch ex As System.Exception
                Console.WriteLine(ex.Message)
            Finally
                SolidEdgeCommunity.OleMessageFilter.Unregister()
            End Try
        End Sub
    End Class
End Namespace
